src/patchelf.cc

```diff
@@ -1129,6 +1141,14 @@ void ElfFile<ElfFileParamNames>::rewriteHeaders(Elf_Addr phdrAddress)
                 dyn->d_un.d_ptr = findSection(".gnu.version_r").sh_addr;
             else if (d_tag == DT_VERSYM)
                 dyn->d_un.d_ptr = findSection(".gnu.version").sh_addr;
+            else if (d_tag == DT_MIPS_RLD_MAP_REL) {
```

Comment:
Could this section be stripped just like `DT_JMPREL` or `DT_REL`?

Reply:
What I am asking is, is it possible to that someone has binaries where this section was stripped.
I fixed a similar issue here: https://github.com/NixOS/patchelf/pull/296

Reply:
@Mic92, thank you for the review.

I understood what you're asking, and I need to give it some thought and run a couple of experiments.

I used to believe that elf with `DT_MIPS_RLD_MAP_REL` in dynamic section but without `.rld_map` is simply broken, but maybe patchelf needs to handle such files. If that's the case, I'll update the code here and add a test for that.

Reply:
Well, elf file with `DT_MIPS_RLD_MAP_REL` in dynamic section but without `.rld_map` is indeed broken: dynamic loader (I've checked glibc 2.32) just writes the debug pointer to the address it calculates from  `DT_MIPS_RLD_MAP_REL` entry. I guess no tool produces such binaries.

But when I just removed `.rtld_map` section with `objcopy`, the resulting binaries mostly worked -- out of sheer luck probably, but still most of them did not crash.

So I rewrote the code to handle such case. Removing `DT_MIPS_RLD_MAP_REL` might have been a  cleaner solution, but `patchelf` does not remove any dynamic section entries yet and  I don't think supporting binaries that were broken in the fist place worth the burden of starting to do so.

And @Mic92, thank you for pointing this case out.